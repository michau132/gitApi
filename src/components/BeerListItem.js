import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Image from './Image';

const BeerListItemStyle = styled.li`
  padding: 20px;
  background-color: #fff;
  box-shadow: 0px 0px 4px 1px rgba(122,122,122,0.9);
  transition: .5s all;
  text-align: center;
  margin: 20px;
  width: 320px;
  cursor: pointer;
  
  :hover {
    transform: scale(1.05);
    box-shadow: 0px 0px 29px 9px rgba(122,122,122,1);
  }
`;

const ImgWrapper = styled.div`
  height: 180px;
  margin-bottom: 10px;
`;

const BeerName = styled.div`
  font-size: 24px;
  color: ${({ theme }) => theme.color.title};
  font-weight: bold;
`;

const BeerDesc = styled.div`
  font-size: 18px;
  color: #b7b7b7;
`;

const BeerListItem = ({
  id, image_url, name, tagline, handleClick,
}) => (
  <BeerListItemStyle
    onClick={() => handleClick(id)}
  >
    <div>
      <ImgWrapper>
        <Image src={image_url} alt={name} />
      </ImgWrapper>
    </div>
    <BeerName>{name}</BeerName>
    <BeerDesc>
      {tagline}
    </BeerDesc>
  </BeerListItemStyle>
);

BeerListItem.propTypes = {
  id: PropTypes.number.isRequired,
  image_url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  tagline: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default BeerListItem;
