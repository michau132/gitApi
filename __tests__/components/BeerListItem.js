import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BeerListItem from '../../src/components/BeerListItem';

describe('<BeerListItem /> component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      id: 1,
      image_url: 'https://via.placeholder.com/150',
      name: 'Beer',
      tagline: 'Drink or die',
      handleClick: jest.fn(),
    };
    wrapper = shallow(<BeerListItem {...props} />);
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('renders Image correctly', () => {
    const img = wrapper.find('Image');
    expect(img.prop('src')).toEqual(props.image_url);
    expect(img.prop('alt')).toEqual(props.name);
  });

  test('firing handleClick ', () => {
    wrapper.simulate('click', { id: 1 });
    expect(props.handleClick).toHaveBeenCalledWith(props.id);
  });
});
