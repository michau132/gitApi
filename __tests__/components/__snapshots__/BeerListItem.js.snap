// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BeerListItem /> component matching snapshot 1`] = `
<styled.li
  onClick={[Function]}
>
  <div>
    <styled.div>
      <Image
        alt="Beer"
        src="https://via.placeholder.com/150"
      />
    </styled.div>
  </div>
  <styled.div>
    Beer
  </styled.div>
  <styled.div>
    Drink or die
  </styled.div>
</styled.li>
`;
